packages/search/src/ago/search.ts

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import { searchItems, ISearchResult } from "@esri/arcgis-rest-items";
+import { IQueryObject, createQueryObject } from "./create-q";
+import { ISearchParams } from "../common/params";
+import { UserSession } from "@esri/arcgis-rest-auth";
+import { encodeAgoQuery } from "./encode-ago-query";
+
+/**
+ * Search for Items in ArcGIS
+ *
+ * @export
+ * @param {IParams} searchRequestOptions
+ * @returns {Promise<ISearchResult>}
+ */
+export function search(
+  params: ISearchParams,
+  authentication: UserSession
+): Promise<ISearchResult> {
```

Comment:
the underlying package `request` supports other requestOptions like passing a custom fetch implementation and custom headers so your signature that strips things down to `params` and `authentication` only is problematic.

@patrickarlt just so happens to have started work on a search param builder of his own in https://github.com/Esri/arcgis-rest-js/pull/509 so we should _really_ try to pool effort.

Reply:
I just took a look at Patrick's work and I think it can work well in building search queries in this library